src/main/java/org/springframework/samples/petclinic/services/PlayerService.java

```diff
@@ -128,4 +130,23 @@ public Boolean existsUser(String username, String email) {
 
 		return false;
 	}
+
+	@Transactional
+    public Player savePlayer(@Valid EditPlayerDto newPlayer) {
+		Player player = new Player();
+		player.setUsername(newPlayer.getUsername());
+        player.setEmail(newPlayer.getEmail());
+		player.setPassword(encoder.encode(newPlayer.getPassword()));
+		this.repository.save(player);
+        return this.repository.save(player);
+    }
```

Comment:
```suggestion
```

Ya existe updatePlayer, por qué es necesario esto? Este método además está mal porque crea un nuevo jugador, con diferente Id y todo

Reply:
Es que este método es para crear el jugador

Reply:
Sé que existe el service createUser, que hay que pasarle un SignupRequest como parámetro, entiendo que eso es para cuando hay que registrarse, he probado usandolo pero no consigo que me funcione y por eso hemos hecho el savePlayer que le pasamos el DTO

Reply:
@SoniaRM Si miráis en swagger el dto, las propiedades son las mismas, a diferencia de que los parámetros son obligatorios, por lo que revisad bien el tema porque o estáis pasando un usuario que ya existe (o cualquiera de las restricciones ya contempladas en el swagger) o no estáis pasando bien un campo que es obligatorio